buildlib/pr/io_demo/az-stage-io-demo.yaml

```diff
@@ -72,14 +81,69 @@ steps:
             ${{ parameters.iodemo_args }}
   displayName: Launch with run_io_demo.sh ( ${{ parameters.name }} )
   timeoutInMinutes: 15
+  condition: eq(variables['single_hca'], 'Yes')
+
+- bash: |
+    set -eEx
+    sudo /hpc/local/bin/lshca
+    source $(workspace)/buildlib/az-helpers.sh
+
+    mkdir -p $(workspace)/${{ parameters.name }}
+    # set UCX environment variables
+    IFS=, read iface1 iface2 <<< ${{parameters.roce_iface}}
+    net1=$(ibdev2netdev | sed -ne 's/\(\w*\) port \([0-9]\) ==> '$iface1' .*/\1:\2/p')
+    net2=$(ibdev2netdev | sed -ne 's/\(\w*\) port \([0-9]\) ==> '$iface2' .*/\1:\2/p')
+
+    IFS=, read host1 host2 <<< $(agent_hosts)
+
+    ip1_h1=$(get_ip ${iface1})
+    ip2_h1=$(get_ip ${iface2})
+    ip1_h2=$(ssh $host2 "source ${PWD}/$(workspace)/buildlib/az-helpers.sh && get_ip ${iface1}" )
+    ip2_h2=$(ssh $host2 "source ${PWD}/$(workspace)/buildlib/az-helpers.sh && get_ip ${iface2}" )
+
+    export UCX_TLS=${{ parameters.iodemo_tls }}
+    export UCX_RNDV_THRESH=4k
+    export UCX_CM_USE_ALL_DEVICES=n
+    export LD_LIBRARY_PATH=$(workspace)/install/lib:$LD_LIBRARY_PATH
+    export UCX_NET_DEVICES=${net1},${net2}
+    # servers
+    IODEMO_ROLE=server_0 \
+      stdbuf -e0 -o0 timeout -s 9 ${{ parameters.duration }} \
+      ${PWD}/$(io_demo_exe) \
+        -d 512:524288 -P 2 -w 16 -t 60 -p 20000 ${{ parameters.iodemo_args }} &>> \
+        $(workspace)/${{ parameters.name }}/server_0.log &
+
+    sleep 5
+    # clients
+
+    ssh ${host2} " \
+      IODEMO_ROLE=client_0 LD_LIBRARY_PATH=${PWD}/$(workspace)/install/lib:$LD_LIBRARY_PATH UCX_NET_DEVICES=${net1} \
+      UCX_CM_USE_ALL_DEVICES=n UCX_RNDV_THRESH=4k \
+        stdbuf -e0 -o0 timeout -s 9 ${{ parameters.duration }} \
+        ${PWD}/$(io_demo_exe) \
+          -d 512:524288 -i 0 -P 2 -w 16 -t 60 ${{ parameters.iodemo_args }} ${ip1_h1}:20000 ${ip2_h1}:20000 &>> \
+        ${PWD}/$(workspace)/${{ parameters.name }}/client_2.log &"
```

Comment:
@avildema Will ```iodemo_analyzer``` parse ```server_0.log``` and ```client_2.log``` file to check whether there're unexpected error?

Reply:
will parce all logs